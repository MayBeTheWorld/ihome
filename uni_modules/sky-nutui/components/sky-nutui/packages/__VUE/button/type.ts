export type ButtonType = 'default' | 'primary' | 'info' | 'success' | 'warning' | 'danger';
export type ButtonSize = 'large' | 'normal' | 'small' | 'mini';
export type ButtonShape = 'square' | 'round';
