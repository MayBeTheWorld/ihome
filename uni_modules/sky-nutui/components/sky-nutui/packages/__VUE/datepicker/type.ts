export type Formatter = (
  type: string,
  option: import('../picker/types').PickerOption
) => import('../picker/types').PickerOption;
export type Filter = (
  columnType: string,
  options: import('../picker/types').PickerOption[]
) => import('../picker/types').PickerOption[];
