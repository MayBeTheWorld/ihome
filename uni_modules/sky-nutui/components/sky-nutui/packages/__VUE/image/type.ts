export type ImageFit = 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
export type ImagePosition = 'center' | 'top' | 'right' | 'bottom' | 'left' | string;
