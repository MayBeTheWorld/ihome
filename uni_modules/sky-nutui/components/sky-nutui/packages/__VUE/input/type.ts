export type InputAlignType = 'left' | 'center' | 'right'; // text-align
export type InputFormatTrigger = 'onChange' | 'onBlur'; // onChange: 在输入时执行格式化 ; onBlur: 在失焦时执行格式化
export type InputType =
  | 'tel'
  | 'url'
  | 'date'
  | 'file'
  | 'text'
  | 'time'
  | 'week'
  | 'color'
  | 'digit'
  | 'email'
  | 'image'
  | 'month'
  | 'radio'
  | 'range'
  | 'reset'
  | 'button'
  | 'hidden'
  | 'number'
  | 'search'
  | 'submit'
  | 'checkbox'
  | 'password'
  | 'textarea'
  | 'datetime-local';

export type InputRule = {
  pattern?: RegExp;
  message?: string;
  required?: boolean;
};

export type ConfirmTextType = 'send' | 'search' | 'next' | 'go' | 'done';
