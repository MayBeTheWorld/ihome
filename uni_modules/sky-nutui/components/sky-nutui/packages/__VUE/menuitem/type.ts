export type MenuItemOption = {
  text: string;
  value: number | string;
};
