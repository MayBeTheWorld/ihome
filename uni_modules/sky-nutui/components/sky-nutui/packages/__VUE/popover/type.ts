export type PopoverTheme = 'light' | 'dark';

export type PopoverLocation =
  | 'bottom'
  | 'top'
  | 'left'
  | 'right'
  | 'top-start'
  | 'top-end'
  | 'bottom-start'
  | 'bottom-end'
  | 'left-start'
  | 'left-end'
  | 'right-start'
  | 'right-end';
