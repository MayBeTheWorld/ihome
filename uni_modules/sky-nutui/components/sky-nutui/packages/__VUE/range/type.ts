export type SliderValue = number | number[];
