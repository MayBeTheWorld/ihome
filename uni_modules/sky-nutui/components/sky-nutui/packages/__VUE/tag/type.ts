export type TagType = 'primary' | 'success' | 'danger' | 'warning';
