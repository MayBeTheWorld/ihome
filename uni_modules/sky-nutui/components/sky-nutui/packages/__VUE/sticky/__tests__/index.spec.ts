import { mount, config, VueWrapper } from '@vue/test-utils';
import { trigger, mockScrollTop } from './../../../utils/unit';
import Sticky from '../index.vue';
import type { ComponentPublicInstance } from 'vue';
import NutButton from '../../button/index.vue';

beforeAll(() => {
  config.global.components = {
    NutButton
  };
});

afterAll(() => {
  config.global.components = {};
});

Object.defineProperty(window.HTMLElement.prototype, 'clientHeight', {
  value: 667
});

function mockStickyRect(wrapper: VueWrapper<ComponentPublicInstance<{}, any>>, rect: Partial<DOMRect>) {
  const mocked = jest.spyOn(wrapper.element, 'getBoundingClientRect').mockReturnValue(rect as DOMRect);

  return () => mocked.mockRestore();
}

test('should sticky to top after scroll', async () => {
  const wrapper = mount(Sticky, {});
  const restore = mockStickyRect(wrapper, {
    top: -100,
    bottom: -90
  });

  await mockScrollTop(1000);
  expect(wrapper.html()).toMatchSnapshot();

  restore();
});

test('should sticky to bottom after scroll', async () => {
  const wrapper = mount(Sticky);
  const restore = mockStickyRect(wrapper, {
    top: 667,
    bottom: 690
  });

  await mockScrollTop(0);
  expect(wrapper.html()).toMatchSnapshot();

  restore();
});
