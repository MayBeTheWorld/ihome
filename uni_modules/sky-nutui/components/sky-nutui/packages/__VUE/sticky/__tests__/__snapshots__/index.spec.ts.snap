// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should sticky to bottom after scroll 1`] = `
"<view>
  <view class=\\"nut-sticky\\"></view>
</view>"
`;

exports[`should sticky to top after scroll 1`] = `
"<view>
  <view class=\\"nut-sticky\\"></view>
</view>"
`;
